### PR TITLE
wine: tweak update script; winePackages.stable: 8.0 -> 8.0.1, winePackages.{unstable,staging}: 8.5 -> 8.7

### PR DESCRIPTION
**Pull request by SFrijters**
###### Description of changes

https://www.winehq.org/announce/8.0.1

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SFrijters** [comment]
Ok, it seems staging 8.6 is broken but fixed upstream.
The nix expression asserts that `assert lib.getVersion wineUnstable == patch.version` so I couldn't just add a different version for staging. Disabling the patchset entirely when there is already a fix seems like a waste, so I opted for this temporary workaround.

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 227785` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>35 packages built:</summary>
  <ul>
    <li>exactaudiocopy</li>
    <li>grapejuice</li>
    <li>grapejuice.dist</li>
    <li>pipelight</li>
    <li>playonlinux</li>
    <li>q4wine</li>
    <li>winbox</li>
    <li>wine (winePackages.full ,winePackages.stableFull)</li>
    <li>wine-staging (winePackages.stagingFull)</li>
    <li>wine-wayland (winePackages.waylandFull)</li>
    <li>wine64 (wine64Packages.full ,wine64Packages.stableFull)</li>
    <li>wine64Packages.base (wine64Packages.stable)</li>
    <li>winePackages.fonts (wine64Packages.fonts ,wineWowPackages.fonts)</li>
    <li>wine64Packages.minimal</li>
    <li>wine64Packages.staging</li>
    <li>wine64Packages.stagingFull</li>
    <li>wine64Packages.unstable</li>
    <li>wine64Packages.unstableFull</li>
    <li>wine64Packages.waylandFull</li>
    <li>winePackages.base (winePackages.stable)</li>
    <li>winePackages.minimal</li>
    <li>winePackages.staging</li>
    <li>winePackages.unstable</li>
    <li>winePackages.unstableFull</li>
    <li>wineWowPackages.base (wineWowPackages.stable)</li>
    <li>wineWowPackages.full (wineWowPackages.stableFull)</li>
    <li>wineWowPackages.minimal</li>
    <li>wineWowPackages.staging</li>
    <li>wineWowPackages.stagingFull</li>
    <li>wineWowPackages.unstable</li>
    <li>wineWowPackages.unstableFull</li>
    <li>wineWowPackages.waylandFull</li>
    <li>wineasio</li>
    <li>yabridge</li>
    <li>yabridgectl</li>
  </ul>
</details>

File: pkgs/applications/emulators/wine/sources.nix
Comment (rhendric) [inline pkgs/applications/emulators/wine/sources.nix:119]: Shouldn't we have `version = "8.6.1";` here instead of the `rev` hack below? The way you have it, the Nix version of wine-staging will be 8.6, despite using the 8.6.1 release.

Comment (SFrijters) [reply pkgs/applications/emulators/wine/sources.nix:119]: Yeah... I'm not too happy about it either way. Originally I just wanted to update stable because that is the only wine I use myself at the moment, so I'm also happy to strip the 8.6 update entirely and wait for 8.7.

Comment (rhendric) [reply pkgs/applications/emulators/wine/sources.nix:119]: What if we changed the problematic assert to check that the versions are equal *or* `lib.hasPrefix "${lib.getVersion wineUnstable}." patch.version`? That might be closer to the true meaning of staging versions.

**Comment by risicle** [comment]
`nixpkgs-review` reveals no new failures on macos 10.15

**Comment by reckenrode** [comment]
I bungled the `nixpkgs-review` on my x86_64-darwin machine, so it’s rerunning. The 8.0.1 changes LGTM.

If you drop 8.6 to wait for 8.7, it seems worth including 2fa8e8d16e82ddbce41698ba3d76b6467460ec15 to facilitate future updates with divergent Gecko versions.

File: pkgs/applications/emulators/wine/sources.nix
Comment (SFrijters) [reply pkgs/applications/emulators/wine/sources.nix:119]: I've done my fair share of wine updates and I've never had to violate the assert before, so I think it's a very rare occurence. I'm happy to be overruled but my first take was to leave the assert alone.

**Comment by SFrijters** [comment]
Stripped the unstable/staging updates for now; chances are the problem resolves itself (in 8.7) at the end of the week and a separate PR can be made then. The update script should correctly update gecko then as well.

**Review by rhendric — APPROVED** [review]
LGTM; wine-stable tested on x86_64-linux and the update script changes are good.

**Comment by SuperSamus** [comment]
Wine 8.7 is out.

**Comment by SFrijters** [comment]
Added new attempt at bumping unstable/staging.

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 227785` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>airwave</li>
  </ul>
</details>
<details>
  <summary>37 packages built:</summary>
  <ul>
    <li>exactaudiocopy</li>
    <li>grapejuice</li>
    <li>grapejuice.dist</li>
    <li>pipelight</li>
    <li>playonlinux</li>
    <li>q4wine</li>
    <li>vkd3d</li>
    <li>vkd3d-proton</li>
    <li>winbox</li>
    <li>wine (winePackages.stableFull)</li>
    <li>wine-staging (winePackages.stagingFull)</li>
    <li>wine-wayland (winePackages.waylandFull)</li>
    <li>wine64 (wine64Packages.stableFull)</li>
    <li>wine64Packages.base (wine64Packages.stable)</li>
    <li>winePackages.fonts (wine64Packages.fonts ,wineWowPackages.fonts)</li>
    <li>wine64Packages.minimal</li>
    <li>wine64Packages.staging</li>
    <li>wine64Packages.stagingFull</li>
    <li>wine64Packages.unstable</li>
    <li>wine64Packages.unstableFull</li>
    <li>wine64Packages.waylandFull</li>
    <li>winePackages.base (winePackages.stable)</li>
    <li>winePackages.minimal</li>
    <li>winePackages.staging</li>
    <li>winePackages.unstable</li>
    <li>winePackages.unstableFull</li>
    <li>wineWowPackages.base (wineWowPackages.stable)</li>
    <li>wineWowPackages.full (wineWowPackages.stableFull)</li>
    <li>wineWowPackages.minimal</li>
    <li>wineWowPackages.staging</li>
    <li>wineWowPackages.stagingFull</li>
    <li>wineWowPackages.unstable</li>
    <li>wineWowPackages.unstableFull</li>
    <li>wineWowPackages.waylandFull</li>
    <li>wineasio</li>
    <li>yabridge</li>
    <li>yabridgectl</li>
  </ul>
</details>

**Comment by SFrijters** [comment]
Accidentally gc'ed the non-free source of `airwave`, so that broke in the review...

**Comment by SuperSamus** [comment]
Wine 8.8 released.
https://www.winehq.org/announce/8.8
(No staging yet.)

**Comment by SFrijters** [comment]
This is why I stopped doing staging updates tbh, it's rather frustrating to not get approval/a merge over weeks while multiple new versions are released (and having to run nixpkgs-review for hours for each release)...
I will be a taking a break from PRs for a while in general, so I'll leave this to someone else for now.

**Comment by reckenrode** [comment]
Thanks for your work on this. I’m interested in the 8.0.1 release since it affects Darwin on macOS 13.3+ on Rosetta, so I’d like to pick this up.

**Comment by SuperSamus** [comment]
Wine 8.9 released (with a Mono update).

**Comment by reckenrode** [comment]
I still want to pick this up, but my time currently is being spent on the Darwin stdenv bump to a newer LLVM. Once that gets to a place where I can flip the PR to ready, I want to look at this next (assuming no one else has picked it up).

**Comment by SuperSamus** [comment]
By the way, I attempted to update to 8.9, and staging wasn't building. I checked, and... staging's latest version is 8.9.1.
Waiting for 8.10, then.

**Pull request closed without merging**
